tests/fixture/testnet/local/network.go

```diff
@@ -632,3 +649,65 @@ func (ln *LocalNetwork) ReadAll() error {
 	}
 	return ln.ReadNodes()
 }
+
+func (ln *LocalNetwork) AddLocalNode(w io.Writer, node *LocalNode, isEphemeral bool) (*LocalNode, error) {
+	// Assume network configuration has been written to disk and is current in memory
+
+	if node == nil {
+		// Set an empty data dir so that PopulateNodeConfig will know
+		// to set the default of `[network dir]/[node id]`.
+		node = NewLocalNode("")
+	}
+
+	// Default to a data dir of [network-dir]/[node-ID]
+	nodeParentDir := ln.Dir
+	if isEphemeral {
+		// For an ephemeral node, default to a data dir of [network-dir]/[ephemeral-dir]/[node-ID]
+		// to provide a clear separation between nodes that are expected to expose stable API
+		// endpoints and those that will live for only a short time (e.g. a node started by a test
+		// and stopped on teardown).
+		//
+		// The data for an ephemeral node is still stored in the file tree rooted at the network
+		// dir to ensure that recursively archiving the network dir in CI will collect all node
+		// data used for a test run.
+		nodeParentDir = filepath.Join(ln.Dir, defaultEphemeralDirName)
+	}
+
+	if err := ln.PopulateNodeConfig(node, nodeParentDir); err != nil {
+		return nil, err
+	}
+
+	// Use dynamic port allocation.
+	var httpPort uint16 = 0
+	var stakingPort uint16 = 0
+
+	// Collect staking addresses of running nodes for use in bootstraping the new node
+	if err := ln.ReadNodes(); err != nil {
+		return nil, fmt.Errorf("failed to read local network nodes: %w", err)
+	}
+	bootstrapIPs := make([]string, 0, len(ln.Nodes))
+	bootstrapIDs := make([]string, 0, len(ln.Nodes))
+	for _, node := range ln.Nodes {
+		if len(node.StakingAddress) == 0 {
+			// Node is not running
+			continue
+		}
+
+		bootstrapIPs = append(bootstrapIPs, node.StakingAddress)
+		bootstrapIDs = append(bootstrapIDs, node.NodeID.String())
+	}
+	if len(bootstrapIDs) == 0 {
+		return nil, errMissingBootstrapNodes
+	}
+
+	node.SetNetworkingConfigDefaults(httpPort, stakingPort, bootstrapIDs, bootstrapIPs)
+
+	if err := node.WriteConfig(); err != nil {
+		return nil, err
+	}
+	if err := node.Start(w, ln.ExecPath); err != nil {
+		return nil, err
+	}
+
+	return node, nil
```

Comment:
```suggestion
	return node, node.Start(w, ln.ExecPath)
```

Reply:
We certainly do not follow that in avalanchego. AvalancheGo expects the caller to verify that the returned error is nil. If it is non-nil, there are no assumptions made about the other returned values.